addons/sale_project/static/tests/project_task_model.js

```diff
@@ -0,0 +1,23 @@
+import { fields } from "@web/../tests/web_test_helpers";
+import { projectModels } from "@project/../tests/project_models";
+import { SaleOrderLine } from "@sale/../tests/mock_server/mock_models/sale_order_line";
+
+
+export class ProjectTask extends projectModels.ProjectTask {
+    _name = "project.task";
+
+    sale_line_id = fields.Many2one({ string: "Sale Order Line", relation: "sale.order.line" });
+}
+
+export class ProjectMilestone extends projectModels.ProjectMilestone {
+    _name = "project.milestone";
+
+    product_uom_qty = fields.Float({ string: "Quantity" });
+    quantity_percentage = fields.Float({ string: "Percentage" });
+}
+
+Object.assign(projectModels, {
+    ProjectMilestone,
+    ProjectTask,
```

Comment:
I don't think you have to re-assign those :thinking:

Reply:
also below that, I don't think it's good practice to add ``SaleOrderLine`` as one of the ``projectModels`` in order to define them together. SaleOrderLine is not really a project model - it'd be better to import and define the model "manually" in each test or create a defineSaleModels inside the sales module that defines the sale models (which we'll import in our tests)

Reply:
> also below that, I don't think it's good practice to add `SaleOrderLine` as one of the `projectModels` in order to define them together. SaleOrderLine is not really a project model - it'd be better to import and define the model "manually" in each test or create a defineSaleModels inside the sales module that defines the sale models (which we'll import in our tests)

I have sent a message to our Vidage server.